test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -961,32 +966,54 @@ private ClusterHealthStatus ensureColor(
             // been removed by the master so that the health check applies to the set of nodes we expect to be part of the cluster.
             .waitForNodes(Integer.toString(cluster().size()));
 
-        ClusterHealthResponse actionGet = clusterAdmin().health(healthRequest).actionGet();
-        if (actionGet.isTimedOut()) {
-            final String hotThreads = clusterAdmin().prepareNodesHotThreads()
-                .setThreads(99999)
-                .setIgnoreIdleThreads(false)
-                .get()
-                .getNodes()
-                .stream()
-                .map(NodeHotThreads::getHotThreads)
-                .collect(Collectors.joining("\n"));
+        final ClusterHealthResponse clusterHealthResponse = clusterAdmin().health(healthRequest).actionGet();
+        if (clusterHealthResponse.isTimedOut()) {
+            final var allocationExplainRef = new AtomicReference<ClusterAllocationExplainResponse>();
+            final var clusterStateRef = new AtomicReference<ClusterStateResponse>();
+            final var pendingTasksRef = new AtomicReference<PendingClusterTasksResponse>();
+            final var hotThreadsRef = new AtomicReference<NodesHotThreadsResponse>();
+
+            final var detailsFuture = new PlainActionFuture<Void>();
+            try (var listeners = new RefCountingListener(detailsFuture)) {
+                clusterAdmin().prepareAllocationExplain().execute(listeners.acquire(allocationExplainRef::set));
+                clusterAdmin().prepareState().execute(listeners.acquire(clusterStateRef::set));
+                clusterAdmin().preparePendingClusterTasks().execute(listeners.acquire(pendingTasksRef::set));
+                clusterAdmin().prepareNodesHotThreads()
+                    .setThreads(9999)
+                    .setIgnoreIdleThreads(false)
+                    .execute(listeners.acquire(hotThreadsRef::set));
```

Comment:
As discussed in other channel, should we remove the hot threads? Or may be limit it to 1 or 2?